cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/store/RemoteRuntimeStoreTest.java

```diff
@@ -0,0 +1,148 @@
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.internal.app.store;
+
+import co.cask.cdap.api.workflow.NodeStatus;
+import co.cask.cdap.common.app.RunIds;
+import co.cask.cdap.internal.app.runtime.ProgramOptionConstants;
+import co.cask.cdap.internal.app.services.http.AppFabricTestBase;
+import co.cask.cdap.internal.app.store.remote.RemoteRuntimeStore;
+import co.cask.cdap.proto.BasicThrowable;
+import co.cask.cdap.proto.Id;
+import co.cask.cdap.proto.ProgramRunStatus;
+import co.cask.cdap.proto.ProgramType;
+import co.cask.cdap.proto.WorkflowNodeStateDetail;
+import co.cask.cdap.proto.id.ProgramRunId;
+import com.google.common.collect.ImmutableMap;
+import com.google.inject.Injector;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Tests implementation of {@link RemoteRuntimeStore}, by using it to perform writes/updates, and then using a
+ * local {@link DefaultStore} to verify the updates/writes.
+ */
+public class RemoteRuntimeStoreTest extends AppFabricTestBase {
+
+  private static DefaultStore store;
+  private static RemoteRuntimeStore runtimeStore;
+
+  @BeforeClass
+  public static void setup() throws Exception {
+    Injector injector = getInjector();
+    store = injector.getInstance(DefaultStore.class);
+    runtimeStore = injector.getInstance(RemoteRuntimeStore.class);
+  }
+
+  @Test
+  public void testSimpleCase() {
+    Id.Program flowId = Id.Flow.from(Id.Namespace.DEFAULT, "test_app", "test_flow");
+    long stopTime = System.currentTimeMillis() / 2000;
+    long startTime = stopTime - 20;
+    String pid = RunIds.generate(startTime * 1000).getId();
+    // to test null serialization (setStart can take in nullable)
+    String twillRunId = null;
+    Map<String, String> runtimeArgs = ImmutableMap.of();
+    Map<String, String> properties = ImmutableMap.of("runtimeArgs", GSON.toJson(runtimeArgs));
+    Map<String, String> systemArgs = ImmutableMap.of("a", "b");
+    RunRecordMeta initialRunRecord =
+      new RunRecordMeta(pid, startTime, null, ProgramRunStatus.RUNNING, properties, systemArgs, twillRunId);
+
+    runtimeStore.setStart(flowId, pid, startTime, twillRunId, runtimeArgs, systemArgs);
+    RunRecordMeta runMeta = store.getRun(flowId, pid);
+    Assert.assertEquals(initialRunRecord, runMeta);
+
+
+    runtimeStore.setSuspend(flowId, pid);
+    Assert.assertEquals(new RunRecordMeta(initialRunRecord, null, ProgramRunStatus.SUSPENDED),
```

Comment:
We don't currently, which is why we separated out Store interface into two different interfaces:
Store and RuntimeStore.
This allows us to keep the remote APIs minimal.